Sources/Jobs/Scheduler/ScheduleBuilder.swift

```diff
@@ -299,6 +299,55 @@ public final class ScheduleBuilder {
     var dayOfWeek: DayOfWeek?
     var time: Time?
     var minute: Minute?
+
+    /// returns the next date that satisfied the created schedule
+    internal func resolveNextDateThatSatisifiesSchedule(date: Date) throws -> Date {
```

Comment:
I thought this was coming in a future PR? Or maybe it's something else? Regardless this method is looking good

Reply:
Again this is where this PR ends essentially. Should I take it out and add it to the next PR? or leave it in?

Reply:
Let's leave it in, I have a better understanding of the line now

Reply:
From our private conversation, I've removed it for now and will add it back in the next PR